Cabal/src/Distribution/Simple/GHC/BuildGeneric.hs

```diff
@@ -399,10 +398,9 @@ gbuild verbosity numJobs pkg_descr lbi bm clbi = do
   -- Determine if program coverage should be enabled and if so, what
   -- '-hpcdir' should be.
   let isCoverageEnabled = exeCoverage lbi
-      distPref = fromFlag $ configDistPref $ configFlags lbi
       hpcdir way
         | gbuildIsRepl bm = mempty -- HPC is not supported in ghci
-        | isCoverageEnabled = toFlag $ Hpc.mixDir distPref way (gbuildName bm)
+        | isCoverageEnabled = toFlag $ Hpc.mixDir (tmpDir </> extraCompilationArtifacts) way
```

Comment:
This is an artifact of `buildGeneric` vs `buildOrRepl` inconsistencies. In one place we have `tmpDir` and in the other `buildDir`. It is just a naming inconsistency between duplicated functions.

(Really, they call it tmpDir because the build dir is postfixed with `-tmp` in some situations like executables. Sam recently opened a ticket about this #9498)